### DISCUSSION TITLE
Getting an error when running the Artillery tests

### DISCUSSION THREAD

**Discussion by ArCiGo** [artilleryio/artillery#1769] [Artillery]
Based on the existing documentation, I attempted to run the following script to execute my tests:

`docker run --platform linux/amd64 -it -v ${PWD}:/performanceTests artilleryio/artillery:latest run /performanceTests/performance.yml`

but I get this error:

`Error: ENOENT: no such file or directory, open '/performanceTests/performance.yml'
    Code: ENOENT
Error: ENOENT: no such file or directory, open '/performanceTests/performance.yml'
ENOENT
ENOENT: no such file or directory, open '/performanceTests/performance.yml'`

I pulled successfully the image, but I'm still getting that error.

This is my current folder structure:

<img width="236" alt="performance yml — YoloGroupDemo 2023-01-16 19-51-42" src="https://user-images.githubusercontent.com/32103862/212793012-18e6598a-4050-443a-a943-a84f2794150b.png">

Is there a solution for this?

**Comment by hassy** [comment]
`ENOENT: no such file or directory, open '/performanceTests/performance.yml'` means that the directory on the host is not being mounted as expected into the Docker container.

If you're running the docker command from the root of your repo, the path to the test script is going to be `/performanceTests/performanceTests/performance.yml`

**Reply by ArCiGo** [reply]
It works! And how can I pass to the script an env variable (in this case, I have my `.env` file that stores the target. I'm using `dotenv-cli`)

**Reply by ArCiGo** [reply]
Nevermind, I found the solution:

`docker run --platform linux/amd64 -it --rm -e SampleURL=https://3dd78cb3-70ab-4481-9209-4372df167733.mock.pstmn.io -v ${PWD}:/performanceTests artilleryio/artillery:latest run /performanceTests/performanceTests/performance.yml`

My Artillery YAML file looks like this:

```YAML
config:
  target: "{{ $processEnvironment.SampleURL }}"
  plugins:
    expect: {}
#rest of the code
```

**Reply by ArCiGo** [reply]
If someone needs it, I left you a Dockerfile to run the Artillery tests (using environmental variables).

```docker
FROM artilleryio/artillery:latest

# Create the folder where our project will be stored
RUN mkdir /my-tests
RUN mkdir /performanceResults

# We make it our work-directory
WORKDIR /my-tests
COPY /performanceResults .

# Let's copy the essential files that we must use to run our scripts. Right value means the destination
COPY ./package.json .
COPY ./package-lock.json .
COPY ./performanceTests ./performanceTests 
COPY ./.env .

# Install the Cypress dependencies in the work directory
# `RUN npm install` was causing a problem at the moment to execute the dockerfile: 'No version of Cypress is installed in: /root/.cache/Cypress/10.3.1/Cypress'
# To solve that, I copied the `package-lock.json .` and I changed to the following command:
RUN npm ci
RUN npm install concurrently

ENTRYPOINT ["npx", "concurrently", "npm:artillery:test"]
```

Since I'm using `dot-env`, my `artillery:test` script looks like this:

```json
"scripts": {
    "artillery:test": "dotenv -- artillery run --output performanceResults/report.json performanceTests/performance.yml && artillery report --output performanceResults/report.html performanceResults/report.json"
  },
```

My `.env` file:

```env
BASE_URL_TARGET=<YourAwesomeURL>
```
